showcase/treemap/tree-tools.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Find branch in tree by omen pair - key-value
+ * @author Akiyamka <akiyamka@gmail.com>
+ *
+ * @param {Object} omen - how to find branch. Example - { id: 123456 }
+ * @param {Object} options [{children: 'children', returnChild: true}];
+ * @param {Boolean} options.returnChild - return whole branch or not
+ * if true return branch started from found child,
+ * @returns {Object} if options.returnChild true - return branch started from found child,
+ * else - return whole branch where child founded.
+ */
+export function findBranchByOmen(omen, options) {
+  const [[omenProp, omenValue]] = Object.entries(omen);
+  const opt = {
+    children: 'children',
+    returnChild: true,
+    ...options
+  };
+
+  function _findBranch(tree) {
+    if (omenValue === tree[omenProp]) {
+      return tree;
+    }
+
+    /* Needed for recursive reduce and find */
+    if (!tree[opt.children]) {
+      return false;
+    };
+
+    return opt.returnChild
+      ? tree[opt.children].reduce((result, branch) => _findBranch(branch) || result, false)
+      : tree[opt.children].find(branch => _findBranch(branch));
+  }
+
+  return data => _findBranch(data);
+}
+
+/**
+ * Fold children, and recalculate weights on leaves.
+ * Used to create tree slice of the required depth
+ * @author Akiyamka <akiyamka@gmail.com>
```

Comment:
again don't include yourself, if someone wants to identify that you did something they can check the blame